model/DataObject.php

```diff
@@ -1926,13 +1926,21 @@ public function many_many_extraFields($component = null) {
 				$manyMany = $SNG_class->stat('belongs_many_many');
 				$candidate = (isset($manyMany[$component])) ? $manyMany[$component] : null;
 				if($candidate) {
+					$relationName = null;
+					// Extract class and relation name from dot-notation
+					if(strpos($candidate, '.') !== false) {
+						list($candidate, $relationName) = explode('.', $candidate);
+					}
+
 					$SNG_candidate = singleton($candidate);
 					$candidateManyMany = $SNG_candidate->stat('many_many');
 
 					// Find the relation given the class
-					if($candidateManyMany) foreach($candidateManyMany as $relation => $relatedClass) {
-						if($relatedClass == $class) {
-							$relationName = $relation;
+					if( ! $relationName && $candidateManyMany) {
+						foreach($candidateManyMany as $relation => $relatedClass) {
+							if($relatedClass == $class) {
```

Comment:
Well, both occurrences are existing code that I’ve just wrapped in `if`/`else` statements.

However, I’m trying to wrap my head around exactly why the `is_subclass_of()` is present _at all_... As far as I can make out, this is the only data structure that it would apply to:

```php
class Team extends DataObject {
    private static $many_many = array('Players' => 'SubPlayer');
}

class Player extends DataObject {
    // Implicit belongs_many_many?
}

class SubPlayer extends Player {
    private static $belongs_many_many = array('Teams' => 'Team');
}

singleton('Player')->many_many('Teams'); // null
```

But I’m sure this would never get past line 2003... I’m so confused!